component/vshn_minio.jsonnet

```diff
@@ -0,0 +1,117 @@
+  'rootUser',
+  'rootPassword',
+];
+
+local minioPlans = common.FilterDisabledParams(minioParams.plans);
+
+local xrd = xrds.XRDFromCRD(
+  'xvshnminios.vshn.appcat.vshn.io',
+  xrds.LoadCRD('vshn.appcat.vshn.io_vshnminios.yaml', params.images.appcat.tag),
+  defaultComposition='vshnminio.vshn.appcat.vshn.io',
+  connectionSecretKeys=connectionSecretKeys,
+) + xrds.WithPlanDefaults(minioPlans, minioParams.defaultPlan);
+
+local resizeServiceAccount = kube.ServiceAccount('sa-sts-deleter') + {
+  metadata+: {
+    namespace: params.services.controlNamespace,
+  },
+};
+
+local resizeClusterRole = kube.ClusterRole('appcat:job:minio:resizejob') {
+  rules: [
+    {
+      apiGroups: [ 'helm.crossplane.io' ],
+      resources: [ 'releases' ],
+      verbs: [ 'get', 'list', 'watch', 'update', 'patch', 'create', 'delete' ],
+    },
+    {
+      apiGroups: [ 'apps' ],
+      resources: [ 'statefulsets' ],
+      verbs: [ 'delete', 'get', 'watch', 'list', 'update', 'patch' ],
+    },
+    {
+      apiGroups: [ 'helm.crossplane.io' ],
+      resources: [ 'releases' ],
+      verbs: [ 'update', 'get' ],
+    },
+    {
+      apiGroups: [ '' ],
+      resources: [ 'pods' ],
+      verbs: [ 'list', 'get', 'update', 'delete' ],
+    },
+  ],
+};
+
+local resizeClusterRoleBinding = kube.ClusterRoleBinding('appcat:job:minio:resizejob') + {
+  roleRef_: resizeClusterRole,
+  subjects_: [ resizeServiceAccount ],
+};
+
+local composition =
+  kube._Object('apiextensions.crossplane.io/v1', 'Composition', 'vshnminio.vshn.appcat.vshn.io') +
+  common.SyncOptions +
+  common.VshnMetaVshn('Minio', 'distributed', 'true', minioPlans) +
+  {
+    spec: {
+      compositeTypeRef: comp.CompositeRef(xrd),
+      writeConnectionSecretsToNamespace: minioParams.secretNamespace,
+      functions:
+        [
+          {
+            name: 'minio-func',
+            type: 'Container',
+            config: kube.ConfigMap('xfn-config') + {
+              metadata: {
+                labels: {
+                  name: 'xfn-config',
+                },
+                name: 'xfn-config',
+              },
+              data: {
+                imageTag: common.GetAppCatImageTag(),
+                minioChartRepository: params.charts.minio.source,
+                minioChartVersion: params.charts.minio.version,
+                plans: std.toString(minioPlans),
+                defaultPlan: minioParams.defaultPlan,
+              },
+            },
+            container: {
+              image: 'minio',
+              imagePullPolicy: 'IfNotPresent',
+              timeout: '20s',
+              runner: {
+                endpoint: minioParams.grpcEndpoint,
+              },
+            },
+          },
+        ],
+    },
+  };
+
+if params.services.vshn.enabled && minioParams.enabled then {
+  '20_xrd_vshn_minio': xrd,
+  '20_rbac_vshn_minio': xrds.CompositeClusterRoles(xrd),
+  '20_rbac_vshn_minio_resize': [ resizeClusterRole, resizeServiceAccount, resizeClusterRoleBinding ],
```

Comment:
I'm a bit confused about this resize service account? It doesn't seem to be referenced anywhere else?

Reply:
yeah I removed it again. Was a copy&paste thing